schneiderWallbox/integrationpluginschneiderwallbox.cpp

```diff
@@ -0,0 +1,546 @@
+    }
+
+    qCDebug(dcSchneiderElectric()) << "Starting network discovery...";
+    NetworkDeviceDiscoveryReply *discoveryReply = hardwareManager()->networkDeviceDiscovery()->discover();
+    connect(discoveryReply, &NetworkDeviceDiscoveryReply::finished, info, [=](){
+        qCDebug(dcSchneiderElectric()) << "Discovery finished. Found" << discoveryReply->networkDeviceInfos().count() << "devices";
+        foreach (const NetworkDeviceInfo &networkDeviceInfo, discoveryReply->networkDeviceInfos()) {
+            qCDebug(dcSchneiderElectric()) << networkDeviceInfo;
+
+            if (!networkDeviceInfo.macAddressManufacturer().contains("TELEMECANIQUE", Qt::CaseSensitivity::CaseInsensitive)) {
+                continue;
+            }
+
+            ThingClass thingClass = supportedThings().findById(info->thingClassId());
+            ParamTypeId macAddressParamType = thingClass.paramTypes().findByName("mac").id();
+
+            ThingDescriptor descriptor(info->thingClassId(), thingClass.displayName(), networkDeviceInfo.address().toString());
+            descriptor.setParams({Param(macAddressParamType, networkDeviceInfo.macAddress())});
+
+            // Check if we already have set up this device
+            Thing *existingThing = myThings().findByParams(descriptor.params());
+            if (existingThing) {
+                qCDebug(dcSchneiderElectric()) << "Found already existing" << thingClass.name() << "wallbox:" << existingThing->name() << networkDeviceInfo;
+                descriptor.setThingId(existingThing->id());
+            } else {
+                qCDebug(dcSchneiderElectric()) << "Found new" << thingClass.name() << "wallbox";
+            }
+
+            info->addThingDescriptor(descriptor);
+        }
+        info->finish(Thing::ThingErrorNoError);
+    });
+}
+
+
+void IntegrationPluginSchneiderWallbox::setupThing(ThingSetupInfo *info)
+{
+    Thing *thing = info->thing();
+
+    /* Probably not needed.
+    if (thing->thingClassId() != schneiderEVlinkThingClassId) {
+        qCWarning(dcSchneiderElectric()) << "Could not setup thing: unhandled device class" << thing->thingClass();
+        info->finish(Thing::ThingErrorThingClassNotFound);
+        return;
+    }
+    */
+
+    if (m_schneiderDevices.contains(thing)) {
+        qCDebug(dcSchneiderElectric()) << "Reconfiguring existing thing" << thing->name();
+        m_schneiderDevices.take(thing)->deleteLater();
+    } else {
+        qCDebug(dcSchneiderElectric()) << "Setting up a new device:" << thing->params();
+    }
+
+    MacAddress mac = MacAddress(thing->paramValue("mac").toString());
+    if (!mac.isValid()) {
+        info->finish(Thing::ThingErrorInvalidParameter, QT_TR_NOOP("The given MAC address is not valid."));
+        return;
+    }
+    NetworkDeviceMonitor *monitor = hardwareManager()->networkDeviceDiscovery()->registerMonitor(mac);
+
+    uint port = thing->paramValue("port").toUInt();
+    quint16 slaveId = thing->paramValue("slaveId").toUInt();
```

Comment:
Can port and slaveID be changed on the device? Normally they can't be changed, in which case I would just drop the paramTypes and hardcode the values in here... If they can be changed, then indeed, this is required.